evm/chains/tester/__init__.py

```diff
@@ -42,91 +48,76 @@ def create_header_from_parent(cls,
-
-
-class FrontierTesterVM(MaintainGasLimitMixin, BaseFrontierVM):  # type: ignore
-    pass
-
-
-class BaseHomesteadTesterVM(MaintainGasLimitMixin, BaseHomesteadVM):  # type: ignore
-    pass
-
 
-class TangerineWhistleTesterVM(MaintainGasLimitMixin, BaseTangerineWhistleVM):  # type: ignore
-    pass
-
-
-class SpuriousDragonTesterVM(MaintainGasLimitMixin, BaseSpuriousDragonVM):  # type: ignore
-    pass
-
-
-INVALID_FORK_ACTIVATION_MSG = (
-    "The {0}-fork activation block may not be null if the {1}-fork block "
-    "is non null"
+MAINNET_VMS = collections.OrderedDict(
+    (vm.fork, type(vm.__name__, (MaintainGasLimitMixin, vm), {}))
+    for _, vm
+    in MainnetChain.vms_by_range.items()
 )
 
-
-@reversed_return
-def _generate_vm_configuration(homestead_start_block=None,
-                               dao_start_block=None,
-                               tangerine_whistle_start_block=None,
-                               spurious_dragon_block=None):
-    # If no explicit configuration has been passed, configure the vm to start
-    # with the latest fork rules at block 0
-    no_declared_blocks = (
-        spurious_dragon_block is None and
-        tangerine_whistle_start_block is None and
-        homestead_start_block is None
+ForkStartBlocks = Sequence[Tuple[int, Union[str, Type[BaseVM]]]]
+VMStartBlock = Tuple[int, Type[BaseVM]]
+
+
+@to_tuple
+def _generate_vm_configuration(*fork_start_blocks: ForkStartBlocks,
+                               dao_start_block: Union[int, bool]=None,
+                               ) -> Generator[VMStartBlock, None, None]:
+    # if no configuration was passed in, initialize the chain with the *latest*
+    # Mainnet VM rules active at block 0.
+    if not fork_start_blocks:
+        yield (0, last(MAINNET_VMS.values()))
+        return
+
+    fork_names = set(
+        fork_name for
+        _, fork_name
+        in fork_start_blocks
+        if isinstance(fork_name, str)
     )
-    if no_declared_blocks:
-        yield (0, SpuriousDragonTesterVM)
-
-    if spurious_dragon_block is not None:
-        yield (spurious_dragon_block, SpuriousDragonTesterVM)
+    unknown_forks = sorted(fork_names.difference(
+        MAINNET_VMS.keys()
+    ))
+    if unknown_forks:
+        raise ValidationError("Configuration contains unknown forks: {0}".format(unknown_forks))
+
+    if dao_start_block is not None and 'homestead' not in fork_names:
+        raise ValidationError(
+            "The `dao_start_block` parameter is only valid for the 'homestead' "
+            "fork rules.  The 'homestead' VM was not included in the provided "
+            "fork configuration"
+        )
 
-        remaining_blocks_not_declared = (
-            homestead_start_block is None and
-            tangerine_whistle_start_block is None
+    # Validate that there is at least one start block for block 0
+    start_blocks = sorted(set(start_block for start_block, _ in fork_start_blocks))
```

Comment:
why `sorted`?

Reply:
misplaced.  moving it to within the `if` block into the error message format.